test/jdk/jdk/jfr/api/consumer/TestChunkInputStreamAvailable.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2021, Alibaba Group Holding Limited. All Rights Reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation. Alibaba designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+/**
+ * @test TestChunkInputStreamAvailable
+ * @key jfr
+ * @requires vm.hasJFR
+ * @library /test/lib
+ * @run main/othervm jdk.jfr.api.consumer.TestChunkInputStreamAvailable
+ */
+package jdk.jfr.api.consumer;
+
+import java.io.File;
+import java.io.InputStream;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+
+import jdk.jfr.FlightRecorder;
+import jdk.jfr.Recording;
+import jdk.test.lib.Asserts;
+import jdk.test.lib.jfr.EventNames;
+
+public class TestChunkInputStreamAvailable {
+
+    public static void main(String[] args) throws Exception {
+        try (Recording r = new Recording()) {
+            r.enable(EventNames.JavaThreadStatistics)
+             .withPeriod(Duration.ofMillis(100));
+            r.start();
+            File repository = Path.of(System.getProperty("jdk.jfr.repository")).toFile();
```

Comment:
I would prefer a simpler and more deterministic test that executes quickly and doesn't involve timing or other events (to reduce false positives).

How about this? (haven't tried it).

```
        public static void main(String... args) throws Exception {
        try (Recording r = new Recording()) {
            r.start();
            try (Recording s = new Recording()) {
                s.start(); // rotate
                s.stop(); // rotate
            }
            r.stop();
            try (InputStream is = r.getStream(null, null)) {
                int left = is.available();
                if (left != r.getSize()) {
                    String msg = "Expected IS::available() " + left;
                    msg += " to equal recording size " + r.getSize();
                    throw new Exception(msg);
                }
                while (is.read() != -1) {
                    left--;
                    int available = is.available();
                    if (available != left) {
                        String msg = "Expected IS::available() to return " + left;
                        msg += ", got " + available;
                        throw new Exception(msg);
                    }
                }
            }
        }
    }
```

Reply:
Make sense.
Updated.